### PR TITLE
unstable -> failed

### PR DESCRIPTION
**Pull request by lavalamp**
@fejta and @ixdy

### CODE REVIEW

File: hack/jenkins/e2e-runner.sh [hunk 1]
```diff
@@ -192,6 +192,7 @@ function e2e_test() {
         echo "Publish version to ci/latest-green.txt: $(cat version)"
         gsutil cp ./version gs://kubernetes-release/ci/latest-green.txt
     fi
+    return ${exitcode}
```
Comment (ixdy) [inline hack/jenkins/e2e-runner.sh]: we need to be careful about this, I think - we need to make sure the cluster is torn down before we errexit.

Comment (lavalamp) [inline hack/jenkins/e2e-runner.sh]: That's what I wanted to ask. I will make some more changes.

Comment (lavalamp) [inline hack/jenkins/e2e-runner.sh]: updated, PTAL

File: hack/jenkins/e2e-runner.sh [hunk 2]
```diff
@@ -362,6 +367,8 @@ if [[ "${USE_KUBEMARK:-}" == "true" ]]; then
   # junit.xml results for test failures and not process the exit code.  This is needed by jenkins to more gracefully
   # handle blocking the merge queue as a result of test failure flakes.  Infrastructure failures should continue to
   # exit non-0.
+  # TODO: The above comment is no longer accurate. Need to fix this before
```
Comment (ixdy) [inline hack/jenkins/e2e-runner.sh]: issue # to track?

Comment (lavalamp) [inline hack/jenkins/e2e-runner.sh]: added.

File: hack/jenkins/e2e-runner.sh (hunk 1 above)
Comment (fejta) [inline hack/jenkins/e2e-runner.sh]: What about when --isup fails?

File: hack/jenkins/e2e-runner.sh [hunk 3]
```diff
@@ -192,13 +192,18 @@ function e2e_test() {
...
 }
 
 echo "--------------------------------------------------------------------------------"
 echo "Test Environment:"
 printenv | sort
 echo "--------------------------------------------------------------------------------"
 
+# Set this var instead of exiting-- we must do the cluster teardown step. We'll
+# return this at the very end.
+EXIT_CODE=0
```
Comment (fejta) [inline hack/jenkins/e2e-runner.sh]: Don't we already have `exitcode`? Why do we need a second one?

Comment (fejta) [inline hack/jenkins/e2e-runner.sh]: I guess is just local to the function but seems a little odd.

Comment (lavalamp) [inline hack/jenkins/e2e-runner.sh]: Yeah I dunno what the rules are in shell, but I like my local variables and global ones to not interact...

File: hack/jenkins/e2e-runner.sh (hunk 1 above)
Comment (lavalamp) [inline hack/jenkins/e2e-runner.sh]: I guess that was already triggering a complete test failure? Either that or it wasn't causing a failure at all?

**Comment by fabioy** [comment]
@fejta Could you create the cherrypick PR?

**Pull request merged**